api/v1beta1/conditions.go

```diff
@@ -88,7 +88,7 @@ const (
 
 	// NodeSetDeploymentReadyCondition Status=True condition indicates if the
 	// NodeSet Deployment is finished and successful.
-	NodeSetDeploymentReadyCondition string = "%s Deployment ready"
+	NodeSetDeploymentReadyCondition condition.Type = "NodeSetDeploymentReady"
```

Comment:
I'm confused about the point of this change. NodeSetDeploymentReadyCondition is changed to condition.Type, but then you don't use it anywhere. You switch all the previous uses to use NodeSetDeploymentReadyMessage, which is still a string, as it needs to be so that we can have separate conditions for each NodeSet.

Reply:
Should all of the `*Condition` ones be `condition.Type`:
```
❯ grep Condition api/v1beta1/conditions.go | grep -Ev '^.\/\/'
	ServiceReadyCondition string = "%s service ready"
	SetupReadyCondition condition.Type = "SetupReady"
	RoleBareMetalProvisionReadyCondition condition.Type = "RoleBaremetalProvisionReady"
	NodeSetBareMetalProvisionReadyCondition condition.Type = "NodeSetBaremetalProvisionReady"
	NodeSetIPReservationReadyCondition condition.Type = "NodeSetIPReservationReady"
	NodeSetDNSDataReadyCondition condition.Type = "NodeSetDNSDataReady"
	NodeSetDeploymentReadyCondition condition.Type = "NodeSetDeploymentReady"
	NodeSetServiceDeploymentReadyCondition string = "%s %s Deployment ready"
```

It's more about the inconsistency. I was going to add a test for it, then also realised we weren't using it. But if it's going to exist, it seemed like it should be a `condition.Type`.

Reply:
Ah ServiceReadyCondition is also a string. Seems like that should be a `condition.Type` as well

Reply:
condition.Type is just a string anyway. This PR doesn't fix the inconsistency since it just switches to use the Message (also a string) as the type parameter.

My intent is to customize the condition type based on the NodeSet name or Service name, so we can dynamically report the status. Alternatively, we could use a single condition with a static type string, and then append to the message to indicate the state of different NodeSets or services. Although that makes it much harder to interpret the status programatically.